File: src/Blazor.MaterialUI/MaterialUiJsInterop.cs
using Microsoft.AspNetCore.Blazor.Browser.Interop;

namespace Blazor.MaterialUI
{
    public class MaterialUiJsInterop
    {
        public static string AppendStyle(string css)
        {
            return RegisteredFunction.Invoke<string>(
                "Blazor.MaterialUI.ExampleJsInterop.AppendStyle",
                css);
        }
    }
}

File: src/Blazor.MaterialUI/content/materialUiJsInterop.js
// This file is to show how a library package may provide JavaScript interop features
// wrapped in a .NET API

Blazor.registerFunction('Blazor.MaterialUI.ExampleJsInterop.AppendStyle', function (css) {
    head = document.head || document.getElementsByTagName('head')[0],
        style = document.createElement('style');

    style.type = 'text/css';
    if (style.styleSheet) {
        style.styleSheet.cssText = css;
    } else {
        style.appendChild(document.createTextNode(css));
    }

    head.appendChild(style);
});
